src/Illuminate/Collections/Traits/EnumeratesValues.php

```diff
@@ -453,6 +453,21 @@ public function sum($callback = null)
         }, 0);
     }
 
+    /**
+     * Get the first item by the given key value pair, but only if
+     * exactly one item matches the criteria. Otherwise, throw
+     * an exception.
+     *
+     * @param  string  $key
+     * @param  mixed  $operator
+     * @param  mixed  $value
+     * @return mixed
+     */
+    public function soleWhere($key, $operator = null, $value = null)
+    {
+        return $this->sole($this->operatorForWhere(...func_get_args()));
+    }
```

Comment:
Since, unlike the `first` method, `sole` doesn't take a `$default` parameter , there's no need for a separate `soleWhere` method. Constraints can be passed into the `sole` method directly, like `contains`.

Reply:
That makes sense. So do you suggest removed the ` soleWhere ` method completely?